lua/autorun/server/sv_pvp_movespeed.lua

```diff
@@ -83,6 +96,23 @@ local function getPlayerWeight( ply )
     return totalWeight
 end
 
+local function dropPlyWeapon( ply )
+    local currentWeapon = ply:GetActiveWeapon()
+    if isUndroppable[currentWeapon:GetClass()] then
+        ply:ChatPrint("This weapon is unable to be dropped!")
+        return
+    end
+
+    ply:DropWeapon( currentWeapon )
+    
+    currentWeapon.despawn = timer.Simple( 10, function()
+        if not IsValid( currentWeapon ) then return end
+        if IsValid( currentWeapon.Owner ) then return end
```

Comment:
Weapons do not retain their owner when dropped?

Reply:
Its just so the weapon isn't removed when its being held by a player, like if they pick the weapon back up.